agents/postgres/parser/parser.go

```diff
@@ -0,0 +1,109 @@
+package parser
+
+import (
+	"reflect"
+
+	pgquery "github.com/lfittl/pg_query_go"
+	pgquerynodes "github.com/lfittl/pg_query_go/nodes"
+)
+
+type Parser struct{}
+
+// ExtractTables extracts table names from query.
+func (p *Parser) ExtractTables(query string) (tables []string, err error) {
+	tree, err := pgquery.Parse(query)
+	if err != nil {
+		return nil, err
```

Comment:
Please use `errors.Wrap` there.

Let's also add `recover()` to this function and convert panic to error with stack trace. Then I will remove `recover()` in pg_query_go: https://github.com/lfittl/pg_query_go/blob/19d3aedd79fd454acc861586bb95483ed7d053c3/pg_query.go#L26

Reply:
`recover()` part wasn't done.